src/binder/mod.rs

```diff
@@ -101,6 +102,23 @@ pub struct Binder {
     table_occurrences: HashMap<TableRefId, u32>,
 }
 
+pub fn bind_header(mut chunk: array::Chunk, stmt: Statement) -> array::Chunk {
```

Comment:
I tried it. If I put it in bind_stmt, there may be a lot of changes because the chunk needs to be passed in. When calling, bind_stmt is called first and then the chunk is generated.

Reply:
let stmt_copy = stmt.clone();
            let mut binder = crate::binder::Binder::new(self.catalog.clone());
            let bound = binder.bind(stmt)?;
            let optimized = optimizer.optimize(&bound);
            let executor = match self.storage.clone() {
                StorageImpl::InMemoryStorage(s) => {
                    crate::executor::build(optimizer.clone(), s, &optimized)
                }
                StorageImpl::SecondaryStorage(s) => {
                    crate::executor::build(optimizer.clone(), s, &optimized)
                }
            };
            let output = executor.try_collect().await?;
            let mut chunk = Chunk::new(output);
            chunk = bind_header(chunk, stmt_copy);
            outputs.push(chunk);

Reply:
Maybe we need to think about whether to modify this part like this

Reply:
Maybe I can consider refactoring the binder select, and consider merging this PR first.

Reply:
This function (or `bind_stmt`) can return the header itself only. Then put the header into chunk at the final step.

```rust
let (bound, header) = binder.bind(stmt)?;
// ...
let mut chunk = Chunk::new(output);
chunk.set_header(header);
```

This way we can avoid copy the statement.

Reply:
Yes, then my changes may be a bit big. I will see if I can fix it together with select. This may take some time.

Reply:
I come up with another idea: the header may be inferred from the `bound` plan.
You can try it in the following PR.

Reply:
Let me try it, it's a very interesting challenge